File: backend/src/main/java/com/mysite/knitly/domain/payment/repository/PaymentRepository.java
package com.mysite.knitly.domain.payment.repository;

import com.mysite.knitly.domain.payment.entity.Payment;
import com.mysite.knitly.domain.payment.entity.PaymentStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public interface PaymentRepository extends JpaRepository<Payment, Long> {

    Optional<Payment> findByOrder_OrderId(Long orderId);
    Optional<Payment> findByTossPaymentKey(String tossPaymentKey);

    List<Payment> findByPaymentStatusAndRequestedAtBefore(
            PaymentStatus status,
            LocalDateTime threshold
    );
}

File: backend/src/main/java/com/mysite/knitly/domain/payment/repository/PaymentRepository.kt
package com.mysite.knitly.domain.payment.repository

import com.mysite.knitly.domain.payment.entity.Payment
import com.mysite.knitly.domain.payment.entity.PaymentStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface PaymentRepository : JpaRepository<Payment, Long> {

    fun findByOrder_OrderId(orderId: Long?): Payment?

    fun findByTossPaymentKey(tossPaymentKey: String): Payment?

    fun findByPaymentStatusAndRequestedAtBefore(
        status: PaymentStatus,
        threshold: LocalDateTime
    ): List<Payment>
}
